### PR TITLE
feat(charts): add tls support to helm chart

### PR DESCRIPTION
**Pull request by dmccaffery**
* add tls variable block to configure service and pod with secure-port
* add ability to create cert-manager certificate
* add support for host ports (both http and https)
* add helm test for tls port
* add example values for secure-port deployment
  - this assumes certificate manager is deployed to the cluster
* pull out script blocks into `hack` path
* update e2e workflow to use scripts in `hack`
* install cert manager and self-signed cluster issuer in e2e
* deploy podinfo with secure port and certificate enabled
* add `hack/e2e.sh` script, which can be used to execute the github
  workflow locally

Closes: #107

### CODE REVIEW

**Comment by dmccaffery** [comment]
No idea if pulling out the e2e workflow stages as separate scripts will be acceptable or not, but I took a stab at it just in case. It makes "testing" the workflow easier as `act` did not work at all when installing kind; presumably because dind / dood isn't configured properly (if its even being used).

I am happy to revert if it makes the community frown; just let me know!

File: charts/podinfo/templates/deployment.yaml [hunk 1]
```diff
@@ -87,6 +102,17 @@ spec:
             - name: http
               containerPort: {{ .Values.service.httpPort | default 9898 }}
               protocol: TCP
+              {{- if .Values.service.hostPort }}
+              hostPort: {{ .Values.service.hostPort }}
+              {{- end }}
+            {{- if .Values.tls.enabled }}
+            - name: https
+              containerPort: {{ .Values.tls.port | default 98899 }}
```
Comment (stefanprodan) [inline charts/podinfo/templates/deployment.yaml:110]: Shouldn't this be `default  9899`?

Comment (dmccaffery) [reply charts/podinfo/templates/deployment.yaml]: fixed -- good catch. I blame key ghosting. O.o

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @dmccaffery 🏅

**Pull request merged**